### DISCUSSION TITLE
Letzten Eintrag aus Kalender ziehen

### DISCUSSION THREAD

**Discussion by therealchrismcfly** [neuefische/questions#125] [Web] [answered]
Guten Morgen,

folgende Situation: ich füge eine Übung auf der Startseite zu einem bestimmten Datum im Kalender hinzu. Auf der Plannerseite kann ich dann für jede Karte die Werte Reps, Sets, Weight beliebig für jeden Tag unterschiedlich anpassen, um mein Training zu planen. Nun möchte ich, dass mir auf der Startseite immer der letzte Eintrag der Reps, Sets, Weight angezeigt wird. 

Beispiel: Freitag war der letzte Tag, an dem ich Squats gemacht habe mit 15kg. Also werden mir dann die 15kg in der entsprechenden Karte auf der Startseite angezeigt. Die möchte ich auf der Startseite auch garnicht mehr anpassen, ich möchte immer nur den aktuellen Eintrag angezeigt bekommen. 

Die konkrete Frage ist also: Wie kann ich den Wert eines letzten Kalendereintrags zugreifen?

Hier ist mein aktuelles Repo: https://github.com/therealchrismcfly/capstone-project/tree/datarefactoring

Vielen Dank schonmal <3

**Comment by alexclaes** [comment, marked as answer]
Hi Chris,

es gibt verschiedene Ansätze. Ich denke folgendes könnte bei deinem Code gut funktionieren:

- Liste aller Übungen laden
- Liste aller Workouts laden
- Über die Liste aller Übungen iterieren und je Übung
  - Alle Workouts zu der jeweiligen Übung laden (`.filter()` anhand von `name`)
  - Ergebnis: Liste aller Workouts zu dieser Übung
  - Workouts anhand des Datums absteigend sortieren (siehe Tipp unten)
  - Den ersten Eintrag aus den Workouts laden (das neuste Workout)
  - Daraus Reps, Sets, Weight auslesen und anzeigen

Hier ein Tipp zum Sortieren nach Datum:

```js
const sortedWorkout = workouts.sort((a, b) => {
    return new Date(b.date) - new Date(a.date);
})
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort
